Frameworks/Foundation/NSIndexPath.mm

```diff
@@ -215,54 +205,51 @@ - (NSInteger)section {
 }
 
 /**
- @Status Stub
-*/
-- (NSIndexPath*)indexPathByRemovingLastIndex {
-    UNIMPLEMENTED();
-    return self;
-}
-
-/**
- @Status Stub
- @Notes
-*/
-- (instancetype)init {
-    UNIMPLEMENTED();
-    return StubReturn();
-}
-
-/**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (void)getIndexes:(NSUInteger*)indexes range:(NSRange)positionRange {
-    UNIMPLEMENTED();
+    if (positionRange.location + positionRange.length > [self length]) {
+        [NSException raise:NSRangeException
+                    format:@"-[%s %s]: range {%lu, %lu} beyond bounds (%lu)",
+                           class_getName([self class]),
+                           sel_getName(_cmd),
+                           positionRange.location,
+                           positionRange.length,
+                           [self length]];
+    }
+
+    std::copy(_indexes.cbegin() + positionRange.location, _indexes.cbegin() + positionRange.location + positionRange.length, indexes);
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 + (BOOL)supportsSecureCoding {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return YES;
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (id)initWithCoder:(NSCoder*)decoder {
-    UNIMPLEMENTED();
-    return StubReturn();
+    NSUInteger length = 0;
+    const NSUInteger* indexes = [decoder isKindOfClass:[NSKeyedUnarchiver class]] ?
+                                    reinterpret_cast<const NSUInteger*>([decoder decodeBytesForKey:c_archiveKey returnedLength:&length]) :
+                                    reinterpret_cast<const NSUInteger*>([decoder decodeBytesWithReturnedLength:&length]);
+    return [self initWithIndexes:indexes length:(length / sizeof(NSUInteger))];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (void)encodeWithCoder:(NSCoder*)coder {
-    UNIMPLEMENTED();
+    if ([self length] > 0L) {
```

Comment:
Other encodeWithEncoder implementations had no serialization done for empty objects and then when calling initWithEncoder returning the empty object when there's no data to create from, and I'd made the (possibly unsound) assumption that NSIndexPath followed suit.  Will check on reference platform and update should it be incorrect.